Models/SentenceDecoder.py

```diff
@@ -51,135 +53,138 @@ def __init__(self, vocab_size, hidden_size):
         self.rnn = LSTMCell(input_size=hidden_size, hidden_size=hidden_size)
         self.out = nn.Linear(hidden_size, vocab_size)
 
-    def forward(self, encoder_outputs, captions):
+    def forward(self, captions):
         """
         Write forward function for the LSTM decoder for the show and tell image captioning task.
         Returns predicted captions and its probability.
         """
         # batch_size x seq_len x hidden_size
         embeddings = self.embedding(captions)
         embeddings = torch.cat((torch.zeros(embeddings.shape[0], 1, embeddings.shape[2]), embeddings), dim=1)
+
         # batch_size x seq_len x hidden_size
-        h_t = torch.zeros(encoder_outputs.shape[0], 1, self.hidden_size)
-        c_t = torch.zeros(encoder_outputs.shape[0], 1, self.hidden_size)
+        h_t = torch.zeros(captions[0], 1, self.hidden_size)
```

Comment:
it should be encoder_outputs.shape[0], basically the image features shape [0]

Reply:
by this you are assuming captions and image features have the same shape which might not be true. and captions[0] is the first caption in the list. but i am not sure whether captions.shape[0] works here as well

Reply:
But aren't we just trying to get the batch size here? I meant to write captions.shape[0]. Wouldn't the batch size be the same regardless?

Reply:
not sure about this, and i think we need the image features? i don't think we can just use the shape of the image features, so i thought about it just now, i don't think we can remove encoder outputs. I think you will also need the encoder outputs for your rnn?